pkg/provision/engine.go

```diff
@@ -83,6 +85,25 @@ func (e *Engine) Run(ctx context.Context) error {
 
 	cReservation := e.source.Reservations(ctx)
 
+	after := time.After(5 * time.Minute)
```

Comment:
This entire section logic is wrong. We need to make sure the clean up procedure pauses provisioning it means it has to be triggered inside the for/select loop otherwise the clean up will be running while the provisining is also "not blocked" .

As a solution i suggest the following (pseudo)
```go
cleanUp := make(chan struct{})
c := cron.New() 
c.AddFunc("@midnight", func() {
   cleanUp <- struct{}{}
})

go func() {
   <- time.After(5*time.Minute)
   cleanUP <- struct{}{}
}()

// in the loop you use 
// case <- cleanUp:
   // call CleanupResources() function
```